### ISSUE TITLE
[BUG] IP allocation for /31 p2p link fails in multi-provider context

### ISSUE THREAD

**Issue by jbemmel** [ipspace/netlab#1625]
### Describe the bug

IP address allocation fails in the following case:

### Lab topology
```
---
#
# IP allocation failure for /31 prefix on regular p2p link in multi-provider lab context
#
provider: clab

addressing:
 p2p:
  prefix: 31                 # '30' succeeds

groups:
  _auto_create: True
  core:
    members: [ c1,c2 ]
    provider: clab

  tor:
    members: [ l1 ]
    provider: libvirt        # <- mixed provider causes issues when link type changes to 'lan'

links:
- c1:
  c2:
  pool: p2p                  # fails: Cannot assign host index 2 in ipv4 from prefix 10.1.0.0/31 to node c2
```

Removing ```provider: libvirt``` fixes the issue, as the (unrelated) c1-c2 link is no longer marked as a 'lan' link

### Version
dev branch

**Comment by ipspace** [comment]
Weird one. It only happens if C2 happens to have ID set to 2, otherwise the "sequential" allocation policy kicks in and everything is OK. Will fix, thank you!

**Comment by ipspace** [comment]
Fixed in 124ad99387703fe64559754475d02732ced23a0e
